src/Corax/Utils/GrowableBuffer.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using Sparrow.Server;
+namespace Corax.Utils;
+
+public interface IBufferGrowth
+{
+    public int GetInitialSize(in long initialSize);
+    public int GetNewSize(in int currentSizeInBytes);
+
+    public bool GrowingThresholdExceed(in int count, in int sizeInBytes);
+}
+
+public readonly struct Slowly : IBufferGrowth
+{
```

Comment:
I don't really like the name `Slowly` because it is not really slow. It actually starts with a high 16Mb baseline and grow progressively. Maybe `Progressive`, `ThresholdGrowth` or something on that line would work. 

There are a few other interesting policies that we have seen and probably we should do the work to start changing that in the rest of the codebase:

LinearGrowth: This policy could increase the buffer size by a fixed amount each time the threshold is exceeded. This might be appropriate when the growth is expected to be steady and predictable.

ExponentialGrowth: This policy could start with a smaller initial size and then grow exponentially, doubling the size each time the threshold is exceeded. This might be useful for scenarios where the expected growth is rapid and unpredictable.

BoundedGrowth<T>: This policy would put a bound to the size and could be built on top of the other strategies.

Reply:
The name was terrible :) Thanks for suggestions